src/Desktop/UIAutomation/A11yAutomation.cs

```diff
@@ -110,9 +113,12 @@ public static IEnumerable<DesktopElement> ElementsFromProcessId(int pid, Registr
                 // if not, it will throw an ArgumentException
                 using (var proc = Process.GetProcessById(pid))
                 {
-                    root = UIAutomation.GetRootElement();
-                    matchingElements = FindProcessMatchingChildrenOrGrandchildren(root, pid);
```

Comment:
`FindProcessMatchingChildrenOrGrandchildren` calls into `GetTreeWalker`, which then returns a shared single object. This may be a part of why this needs to be locked. The concerning thing is that the scanner will also find its way into `GetTreeWalker`, and those would be a lot harder to fix. We might do well to have a separate `IUIAutomation` object inside each `TreeWalkerDataContext`. I'll start working on that change

Reply:
Sounds good, I'll revert the locking change and merge in your change once it's ready :)